pkg/streamingpromql/types/fpoint_ring_buffer.go

```diff
@@ -106,6 +106,16 @@ func (b *FPointRingBuffer) ViewUntilSearchingForwards(maxT int64, existing *FPoi
 	return existing
 }
 
+// View returns a view which includes all points in the ring buffer.
+// The returned view is no longer valid if this buffer is modified (eg. a point is added, or the buffer is reset or closed).
+func (b *FPointRingBuffer) View(existing *FPointRingBufferView) *FPointRingBufferView {
```

Comment:
[nit] Perhaps a name like `ViewOfAll` would make the difference compared to the other `View...` methods clearer?